kepler/tasks.py

```diff
@@ -76,12 +77,36 @@ def submit_to_dspace(job, data):
         pkg.metadata = _fgdc_to_mods(get_fgdc(data))
         with tempfile.NamedTemporaryFile(suffix='.zip') as fp:
             pkg.write(fp)
-            handle = sword.submit(current_app.config['SWORD_SERVICE_URL'], fp.name,
-                                  auth=(username, password))
+            handle = sword.submit(current_app.config['SWORD_SERVICE_URL'],
+                                  fp.name, auth=(username, password))
         job.item.handle = handle
         db.session.commit()
 
 
+def get_geotiff_url_from_dspace(job):
+    """Retrieve the GeoTIFF URL from a DSpace Handle.
+
+        .. note:: assumes the OAI-ORE only contains a single TIFF.
+
+        :param job: :class:`~kepler.models.Job`
+    """
+    handle = job.item.handle.replace('http://hdl.handle.net/', '')
+    ore_url = current_app.config['OAI_ORE_URL'] + handle + '/ore.xml'
+    r = requests.get(ore_url)
+    r.raise_for_status()
+    doc = etree.fromstring(r.content)
+    tif_urls = doc.xpath(
+        '//rdf:Description/@rdf:about[contains(., ".tif")]',
+        namespaces={'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'})
+
+    if len(tif_urls) != 1:
+        raise Exception('Expected 1 tiff, found ' + str(len(tif_urls)))
```

Comment:
I used a generic exception with a clear, useful message. I'm happy to switch to a custom Exception class if that's preferred.